### ISSUE TITLE
Mac m1机器上python 3.10安装paddle-bfloat报错

### ISSUE THREAD

**Issue by pangyoki** [PaddlePaddle/Paddle#42125]
### 问题描述 Issue Description

如题，在Mac m1机器上，python3.10（universal版本）单独安装paddle-bfloat会报错，如下图。初步测试linux、windows、非m1的mac机器上，python3.10可以正常安装paddle-bfloat。

![图片](https://user-images.githubusercontent.com/26408901/164625614-45bc3775-1b47-44f6-9a97-f62a2ca88eb8.png)

看起来是pypi上没有相应的预编译包，默认去拉tar.gz本地编译安装，可能暂时不支持mac m1环境下的编译安装。

paddle准备支持python3.10，依赖paddle-bfloat对mac m1机器python3.10的支持，希望能高优解决这个问题。

### 版本&环境信息 Version & Environment Information

Mac m1机器，python3.10 universal （https://www.python.org/ftp/python/3.10.4/python-3.10.4-macos11.pkg）

**Comment by Aganlengzi** [comment]
@lidanqing-intel @jakpiase hi, in addition, please upload the pre-compiled packages of other tags to avoid local compiling, I think we can refer to Pillow https://pypi.org/project/Pillow/#files

**Comment by jakpiase** [comment]
Hi @Aganlengzi, @pangyoki. I have uploaded additional wheels for paddle_bfloat v0.1.2 into pypi: [https://pypi.org/project/paddle-bfloat/0.1.2/#files](https://pypi.org/project/paddle-bfloat/0.1.2/#files). Mac scenario with python 3.10 is also included here, so I think that your problem should be fixed. Please let me know if it resolved your issue, thanks

**Comment by Aganlengzi** [comment]
> Hi @Aganlengzi, @pangyoki. I have uploaded additional wheels for paddle_bfloat v0.1.2 into pypi: https://pypi.org/project/paddle-bfloat/0.1.2/#files. Mac scenario with python 3.10 is also included here, so I think that your problem should be fixed. Please let me know if it resolved your issue, thanks

@jakpiase Thank you very much!
I checked in my Mac with m1 and python 3.10, it succeed:
<img width="586" alt="image" src="https://user-images.githubusercontent.com/5997715/164951809-bbf39137-f352-47fb-82c7-1aa72078b5c3.png">

<img width="590" alt="image" src="https://user-images.githubusercontent.com/5997715/164951824-5d34bcc7-5f2c-4140-ba34-935488c047ab.png">

**Comment by Aganlengzi** [comment]
@jakpiase Hi, which numpy version is used in your pre-compiled wheels? could it align with Paddle's requirements?
Paddle requires numpy>=1.13 in https://github.com/PaddlePaddle/Paddle/blob/develop/python/requirements.txt
but in some case, it reports incompatible problem：
```
2022-04-24 12:38:26 1/1 Test #1264: test_python_bf16_numpy_datatype ...***Failed    2.46 sec
2022-04-24 12:38:26 RuntimeError: module compiled against API version 0xf but this version of numpy is 0xe
2022-04-24 12:38:26 ImportError: numpy.core._multiarray_umath failed to import
2022-04-24 12:38:26 ImportError: numpy.core.umath failed to import
```
https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/5478824/job/14350996
<img width="831" alt="image" src="https://user-images.githubusercontent.com/5997715/165039413-29555906-a4b6-4380-83ec-75bb63d6e0f4.png">

**Comment by jakpiase** [comment]
Hi @Aganlengzi, I have uploaded some wheels with old numpy versions, sorry for that. I have deleted them, now pypi repository should contain only versions which were built with numpy>=1.13

**Comment by Aganlengzi** [comment]
> Hi @Aganlengzi, I have uploaded some wheels with old numpy versions, sorry for that. I have deleted them, now pypi repository should contain only versions which were built with numpy>=1.13

Hi, @jakpiase thanks for quick reply, but the message seems saying that the numpy version you compiled with is higher than installed one(0xf > 0xe)? if this is true, I think you may compile wheels with lower numpy version(at most 1.13)? I am not sure about that.

Tried locally with wheel: paddle_bfloat-0.1.2-cp37-cp37m-manylinux_2_27_x86_64.whl
<img width="835" alt="image" src="https://user-images.githubusercontent.com/5997715/165119892-8e5fcbfd-1cbb-4d88-afeb-5b6aac9c6c96.png">

Another try with newest numpy and wheel: numpy-1.21.6-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl
<img width="832" alt="image" src="https://user-images.githubusercontent.com/5997715/165121525-f732aece-7c4b-4a59-895b-89443926ec83.png">

**Comment by Aganlengzi** [comment]
@lidanqing-intel @jakpiase hi, there is no arm64 package in https://pypi.org/project/paddle-bfloat/0.1.2/#files,
would you please help upload pre-built arm64 package?

**Comment by jakpiase** [comment]
Hi @Aganlengzi, I have uploaded arm64 wheels for python3.8-3.10.
Just FYI I am currently working on newer version of paddle_bfloat package and it should be finished soon. All future versions of paddle_bfloat will cover all types of wheels that are currently supported. New wheels for older linux systems will be added as well. I have managed to fix issues associated with different numpy versions between paddle_bfloat and client system. In future versions paddle_bfloat wheels will match all numpy wheels, so following combinations of python and numpy will be supported:

| Python version  | Numpy versions|
| ------------- |:-------------:|
| 3.6       | 1.13-1.19     |
| 3.7       | 1.15-1.21    |
| 3.8       | 1.18-1.22     |
| 3.9       | 1.20-1.22    |
| 3.10     | 1.22    |

**Comment by Aganlengzi** [comment]
Have seen precompiled packages, reopen if any related issues.

**Comment by shiyutang** [comment]
linux python3.8 pip-22.1 安装paddle-bfloat=0.1.2 报错：

1. python3.8 -m pip install paddle-bfloat==0.1.2 --no-cache
![image](https://user-images.githubusercontent.com/34859558/168226323-64274a02-0052-4270-aeed-3fb9a3c3ab33.png)
2. 网页上下载对应版本，不能安装：
![14c73242e3dfe252c8102f44580858d9](https://user-images.githubusercontent.com/34859558/168226485-0a55ca40-74ee-4f32-80ad-218c46452969.png)

**Comment by Aganlengzi** [comment]
@lidanqing-intel @jakpiase hi, paddle-bfloat==0.1.2 is depended by 2.3, it is very important to keep as many as possible pre-built wheels, please help check, thanks.
BTW, In @shiyutang‘s env, package with the following tag is ok, but can not be found in this site:https://pypi.org/project/paddle-bfloat/0.1.2/#files
<img width="349" alt="image" src="https://user-images.githubusercontent.com/5997715/168228761-534f526f-4556-4372-839b-e6c469ed01fe.png">

**Comment by shiyutang** [comment]
这样之后，就可以了
![image](https://user-images.githubusercontent.com/34859558/168229928-bf89eb5c-0683-4a41-9843-d67fe0732672.png)

**Comment by lidanqing-vv** [comment]
still fixing paddle_bfloat package CIs, compilation failures are fixed, but there is a problem while building paddlepaddle-gpu package with my changes

**Comment by lidanqing-vv** [comment]
@jakpiase Please make sure 0.1.2 has as many tags as possible because paddle2.3 is using paddle_bfloat16 0.1.2. Thanks!

**Comment by jakpiase** [comment]
@Aganlengzi, shiyutang's wheel cannot be found on https://pypi.org/project/paddle-bfloat/0.1.2/#files, because it was deleted by me, since I have built it with newest numpy version and it was not compatible with older ones which was causing errors listed by you above.
Today I have deleted most of the broken wheels from paddle_bfloat v0.1.2 andm I have added more new wheels, to cover a wider range of supported machines. The recently added wheels were built with `manylinux1` standard, which is the oldest possible(supports distros like Centos 5), so all linux machines should work properly now. I have tested that locally, and linux wheels for all python version supported by PaddlePaddle should work correctly without issues like `RuntimeError: module compiled against API version 0xf but this version of numpy is 0xe`. Although I have fixed wheels for linux machines for v0.1.2, wheels published for windows and macos still won't work 100% correctly, but I cannot replace them, because PyPi doesn't allow publishing a wheel which have the same name as previously added one because of security reasons, so the only way to fix all current issues related to paddle_bfloat package is to upgrade it inside PaddlePaddle to version 0.1.7, which is the newest and is fully tested by me and has all wheels working correctly.

All currently published wheels for the two important versions of paddle_bfloat:
- v0.1.2 wheels: https://pypi.org/project/paddle-bfloat/0.1.2/#files
- v0.1.7 wheels: https://pypi.org/project/paddle-bfloat/0.1.7/#files
